corehq/apps/sso/tests/test_request_helpers.py

```diff
@@ -0,0 +1,238 @@
+from testil import eq
+from django.test import TestCase, RequestFactory, override_settings
+
+from corehq.apps.accounting.models import Subscription
+from corehq.apps.domain.models import Domain
+from corehq.apps.sso.exceptions import SingleSignOnError
+from corehq.apps.sso.models import (
+    AuthenticatedEmailDomain,
+    IdentityProvider,
+    TrustedIdentityProvider,
+)
+from corehq.apps.sso.utils.request_helpers import (
+    get_request_data,
+    is_request_using_sso,
+    is_request_blocked_from_viewing_domain_due_to_sso,
+)
+from corehq.apps.users.models import WebUser
+from corehq.apps.sso.tests import generator
+
+
+@override_settings(SAML2_DEBUG=False)
+def test_get_request_data():
+    request = RequestFactory().get('/sso/test')
+    request.META = {
+        'HTTP_HOST': 'test.org',
+        'PATH_INFO': '/sso/test',
+        'SERVER_PORT': '999',
+    }
+    request.POST = {
+        'post_data': 'test',
+    }
+    request.GET = {
+        'get_data': 'test',
+    }
```

Comment:
`get_request_data` is indeed required to process the request into a form that the `python3-saml` library can handle, as all the frameworks format their request objects differently. both are needed otherwise the library throws an error.